WPMedia/engine-theme-sdk
# TMEDIA 234: Remove unused timezone js files

- [x] still need to test getting the actual localeList
- [x] also nested timezone and regions
- [x] Fix github actions test 


## Description
Take in the path for blocks.json and only install locale list shown


## Jira Ticket
- [TMEDIA-234](https://arcpublishing.atlassian.net/browse/TMEDIA-234)

## Acceptance Criteria
- expandable for customers in different timezones and locales than what we have hard-coded
- no significant client-side bundle being shipped to customers due to "all locales inclusion"

## Test Steps

- Link the engine-theme-sdk branch `TMEDIA-234-date-postinstall-delete` locally in the `.env`
- Make sure fusion release is 2.7.4
-  as a sanity check, also make sure you are using node 10 locally
- Change the localeList to only include one or two locales like so to test the removal behavior of the postinstall script:

```json
  "localeList": [
    "ko"
  ],
```

- Run `npx fusion start -f` to regenerate .fusion. Wait until you can see the webpack output and it's built
- Stop the process with ctrl-c 
- Change the docker-compose.yml in the .fusion folder accordingly: 

- remove network_mode 
- add `- '/Users/{*******your file path*************}/engine-theme-sdk:/opt/engine/bundle/linked_modules/@wpmedia/engine-theme-sdk:rw'` with your file-path to engine-theme-sdk (without src path) and with `rw` rather than `ro:cached`

```yml
webpack:
    build:
      context: ..
      dockerfile: ./.fusion/webpack.Dockerfile
      args:
        FUSION_RELEASE: '${FUSION_RELEASE:-latest}'
    restart: always
    command: watch
    container_name: fusion-webpack
    env_file:
      - ../.env
    environment:
      DB_NAME: '${DB_NAME:-Fusion-News-Theme}'
      MONGO_URL: 'mongodb://data:27017/${DB_NAME:-Fusion-News-Theme}'
      PB_MONGODB_URI: 'mongodb://data:27017'
      NODE_ENV: development
      ENVIRONMENT: localhost
      CONTEXT_PATH: '${CONTEXT_PATH:-pf}'
      DEBUG: '${DEBUG:-''fusion:*''}'
      HTTP_ENGINE: 'http://engine:8080'
      HTTP_RESOLVER: 'http://resolver:8080'
      NODE_OPTIONS: '--max_old_space_size=4096'
      CACHE_PROXY_URL: 'http://localhost:password@cache-proxy:8080/cache'
      CACHE_PREFIX: '0803'
      WATCH: null
      POLL: null
      PORT: '8080'
    volumes:
      - '..:/opt/engine/bundle/src:ro,cached'
      - /opt/engine/bundle/src/node_modules
      - 'shared:/opt/engine/bundle/node_modules:rw,delegated'
      - 'fusion_cache_Fusion-News-Theme:/opt/engine/node_modules/.cache:rw,nocopy,delegated'
      - '../.fusion/babel:/opt/engine/bundle/babel:rw,cached'
      - '../.fusion/build:/opt/engine/bundle/build:rw,cached'
      - '../.fusion/dist:/opt/engine/bundle/dist:rw,cached'
      - '../.fusion/generated:/opt/engine/bundle/generated:rw,cached'
      - '/Users/howardj/sites/engine-theme-sdk:/opt/engine/bundle/linked_modules/@wpmedia/engine-theme-sdk:rw'
```

- Rerun `docker-compose -f ./.fusion/docker-compose.yml up` to make the effects of the docker-compose change
- Go to your Docker dashboard 

![Screen Shot 2021-05-28 at 10 39 05](https://user-images.githubusercontent.com/5950956/120009022-2c5f1200-bfa1-11eb-92d1-337197e3074b.png)
![Screen Shot 2021-05-28 at 10 40 55](https://user-images.githubusercontent.com/5950956/120009015-2b2de500-bfa1-11eb-8e25-922f822fe2d8.png)

- Once within the running image, look at the timezone node modules that were installed without running the new engine theme sdk `cd bundle && ls node_modules/timezone`. You will see all locales installed. this is because the engine theme sdk fallback was published canary* which does not have a postinstall script: 

```
/opt/engine/bundle # ls node_modules/timezone/
Africa        MST7MDT.js    en_HK.js      fr_BE.js      nl_BE.js
America       Mexico        en_NZ.js      fr_CA.js      nl_NL.js
Antarctica    PST8PDT.js    en_US.js      fr_CH.js      package.json
Arctic        Pacific       es_AR.js      fr_FR.js      pl_PL.js
Asia          README.md     es_CL.js      gl_ES.js      pt_BR.js
Atlantic      US            es_CO.js      he_IL.js      pt_PT.js
Australia     WET.js        es_CR.js      hi_IN.js      rfc822.js
Brazil        af_ZA.js      es_DO.js      hr_HR.js      ru_RU.js
CET.js        am_ET.js      es_EC.js      hu_HU.js      si_LK.js
CHANGELOG     ast_ES.js     es_ES.js      id_ID.js      sl_SI.js
CST6CDT.js    bg_BG.js      es_GT.js      index.js      sq_AL.js
Canada        bn_BD.js      es_HN.js      it_CH.js      sr_RS.js
Chile         bn_IN.js      es_MX.js      it_IT.js      sv_SE.js
EET.js        ca_ES.js      es_NI.js      ja_JP.js      synopsis.js
EST.js        cs_CZ.js      es_PA.js      ko_KR.js      ta_IN.js
EST5EDT.js    de_AT.js      es_PE.js      loaded.js     uk_UA.js
Etc           de_CH.js      es_PR.js      locales.js    ur_PK.js
Europe        de_DE.js      es_SV.js      lt_LT.js      vi_VN.js
HST.js        el_GR.js      es_UY.js      lv_LV.js      zh_CN.js
Indian        en_AU.js      es_VE.js      ms_MY.js      zh_HK.js
MET.js        en_CA.js      eu_ES.js      nb_NO.js      zh_TW.js
MST.js        en_GB.js      fi_FI.js      nds_DE.js     zones.js

```
- Now, to test the postinstall of the linked engine theme sdk: 

`pwd`
# opt/engine/bundle
`npm i /opt/engine/bundle/linked_modules/@wpmedia/engine-theme-sdk`

- Then, look at the same timezone node modules. You should only see the locale that you designated in the `localeList`. This will mean that the path to the blocks.json has been located and the localelist has dynamically deleted locales from the node modules

```
/opt/engine/bundle # ls node_modules/timezone/
America       EST.js        MET.js        Pacific       locales.js
CET.js        EST5EDT.js    MST.js        WET.js        package.json
CST6CDT.js    Europe        MST7MDT.js    index.js      rfc822.js
EET.js        HST.js        PST8PDT.js    ko_KR.js      zones.js
```

## Effect Of Changes
### Before
_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After
_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
